Remove use of `RefPtr::unsafeGet()` in `KeyframeEffect::activeViewTimeline()`

graouts opened the pull request:
#### b43b137ecd9acf9d13067c63d064e8de756d2980
<pre>
Remove use of `RefPtr::unsafeGet()` in `KeyframeEffect::activeViewTimeline()`
<a href="https://bugs.webkit.org/show_bug.cgi?id=300810">https://bugs.webkit.org/show_bug.cgi?id=300810</a>
<a href="https://rdar.apple.com/162690218">rdar://162690218</a>

Reviewed by Geoffrey Garen and Chris Dumez.

* Source/WebCore/animation/KeyframeEffect.cpp:
(WebCore::KeyframeEffect::processKeyframes):
(WebCore::KeyframeEffect::animationDidTick):
(WebCore::KeyframeEffect::activeViewTimeline const):
(WebCore::KeyframeEffect::activeViewTimeline): Deleted.
* Source/WebCore/animation/KeyframeEffect.h:

Canonical link: <a href="https://commits.webkit.org/301616@main">https://commits.webkit.org/301616@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/885020c232890878ac9ab650731b1006a8e19fb8

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126538 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/46183 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/37106 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/133427 "Built successfully") | [⏳ 🛠 win ](https://ews-build.webkit.org/#/builders/Win-Build-EWS "Waiting in queue, processing has not started yet") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/076284c6-c0a6-4988-8ffe-c980c3c1d805) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/128409 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46817 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54721 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/96299 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🧪 win-tests ](https://ews-build.webkit.org/#/builders/Win-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/ee40b06e-377c-46b3-819d-7b1ed39f8d53) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129486 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37450 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/113167 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/76824 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/a4514218-b9c4-435f-9a86-0a651be648b2) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/36342 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/31349 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/76735 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/107259 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31629 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/135975 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53229 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40933 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/104800 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53715 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/109492 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/104502 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/49982 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/28315 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50641 "Hash 885020c2 for PR 52395 does not build (failure)") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53149 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/58962 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52431 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55765 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54166 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren commented on Source/WebCore/animation/KeyframeEffect.cpp:
```diff
@@ -3161,9 +3161,9 @@ const ViewTimeline* KeyframeEffect::activeViewTimeline()
-    RefPtr viewTimeline = dynamicDowncast<ViewTimeline>(animation->timeline());
-    if (viewTimeline && viewTimeline->currentTime())
-        return viewTimeline.unsafeGet();
+    auto* viewTimeline = dynamicDowncast<ViewTimeline>(animation->timeline());
```
This is perfectly fine, but if you change the function return signature to `RefPtr<ViewTimeline>` you can save one refcount churn while also solving this: https://build.webkit.org/results/Apple-Sequoia-Safer-CPP-Checks/301548@main%20(5666)/scan-build-output/StaticAnalyzer/WebCore/StaticAnalyzerReports/report-KeyframeEffect.cpp-processKeyframes-47-4983b5.html#EndPath

annevk replied:
But then we'd be violating the guidelines, no? As this isn't really the type of function that should return a RefPtr.

graouts replied:
> This is perfectly fine, but if you change the function return signature to `RefPtr<ViewTimeline>` you can save one refcount churn while also solving this: https://build.webkit.org/results/Apple-Sequoia-Safer-CPP-Checks/301548@main%20(5666)/scan-build-output/StaticAnalyzer/WebCore/StaticAnalyzerReports/report-KeyframeEffect.cpp-processKeyframes-47-4983b5.html#EndPath

I believe this is addressed in the updated patch.

annevk commented on Source/WebCore/animation/KeyframeEffect.cpp:
```diff
@@ -2082,7 +2082,7 @@ void KeyframeEffect::animationDidTick()
     updateAcceleratedActions();
 
-    if (RefPtr viewTimeline = activeViewTimeline())
+    if (auto viewTimeline = protectedActiveViewTimeline())
```
I think this clearly shows why we don't want it, no?

cdumez replied:
This change goes against our coding style.

cdumez replied:
My take is that this line should remain as-is and `protectedActiveViewTimeline()` should be named `activeViewTimeline()`. Having the protected naming when there is no "unprotected" variant is not particularly useful.

With what I suggest, our coding style would remain the same (Do not use `protected` functions to initialize local variables).

cdumez reviewed (changes requested):
This change does not actually remove `unsafeGet()`. I think the rename to `protected` here needs some discussion, considering this now violates our coding style (see failure on style bot).

cdumez commented on Source/WebCore/animation/KeyframeEffect.cpp:
```diff
@@ -2082,7 +2082,7 @@ void KeyframeEffect::animationDidTick()
     updateAcceleratedActions();
 
-    if (RefPtr viewTimeline = activeViewTimeline())
+    if (auto viewTimeline = activeViewTimeline())
```
This should remain as `RefPtr`, we prefer not to hide smart pointers behind `auto` in new code.

graouts committed:
Remove use of `RefPtr::unsafeGet()` in `KeyframeEffect::activeViewTimeline()`
https://bugs.webkit.org/show_bug.cgi?id=300810
rdar://162690218

Reviewed by Geoffrey Garen and Chris Dumez.

* Source/WebCore/animation/KeyframeEffect.cpp:
(WebCore::KeyframeEffect::processKeyframes):
(WebCore::KeyframeEffect::animationDidTick):
(WebCore::KeyframeEffect::activeViewTimeline const):
(WebCore::KeyframeEffect::activeViewTimeline): Deleted.
* Source/WebCore/animation/KeyframeEffect.h:

Canonical link: https://commits.webkit.org/301616@main
